pkg/controller/servicemesh/webhooks/validation/memberroll.go

```diff
@@ -107,32 +117,103 @@ func (v *MemberRollValidator) Handle(ctx context.Context, req atypes.Request) at
 	}
 	if !allowed {
 		// check each namespace separately, but only check newly added namespaces
-		namespacesToCheck := sets.NewString(smmr.Spec.Members...)
-
-		if req.AdmissionRequest.Operation == admissionv1.Update {
-			oldSmmr := &maistrav1.ServiceMeshMemberRoll{}
-			err := v.decoder.DecodeRaw(req.AdmissionRequest.OldObject, oldSmmr)
-			if err != nil {
-				logger.Error(err, "error decoding old object in admission request")
-				return admission.ErrorResponse(http.StatusBadRequest, err)
-			}
-			namespacesToCheck.Delete(oldSmmr.Spec.Members...)
+		namespacesToCheck, err := v.findNewlyAddedNamespaces(smmr, req)
+		if err != nil {
+			return admission.ErrorResponse(http.StatusInternalServerError, err)
 		}
 
-		for _, member := range namespacesToCheck.List() {
-			allowed, err := v.isUserAllowedToUpdatePods(common.NewContextWithLog(ctx, logger.WithValues("namespace", member)), req, member)
-			if err != nil {
-				return admission.ErrorResponse(http.StatusInternalServerError, err)
-			}
-			if !allowed {
-				return validationFailedResponse(http.StatusForbidden, metav1.StatusReasonBadRequest, fmt.Sprintf("user '%s' does not have permission to access project/namespace '%s'", req.AdmissionRequest.UserInfo.Username, member))
+		allowed, rejectedNamespaces, err := v.isUserAllowedToUpdatePodsInAllNamespaces(ctx, req, namespacesToCheck)
+		if err != nil {
+			if errors.Is(err, context.DeadlineExceeded) {
+				logger.Error(err, fmt.Sprintf("timeout while performing SAR checks for %d namespaces", len(namespacesToCheck)))
+				return admission.ErrorResponse(http.StatusBadRequest, fmt.Errorf("too many namespaces in ServiceMeshMemberRoll; validating webhook couldn't perform the authorization checks for all namespaces; either try the operation again as a cluster admin, or add fewer namespaces in a single operation"))
 			}
+			return admission.ErrorResponse(http.StatusInternalServerError, err)
```

Comment:
We probably want to log the error again here, will make future debugging easier

Reply:
You're absolutely right. I've fixed this here and in one other place, where we also returned a 500 error.